roachtest: fix failover disk stall

DarrylWong opened the pull request:
This fixes the failover disk staller to be compatible with the FI library. Specifically, it does not attempt to recover in cleanup, as this is already handled in the FI framework.

Additionally, it temporarily disables concurrent disk stall failures from being injected as this is an invariant of the FI framework. Instead, they should be injected in one shot.

Fixes: #148739
Fixes: #148742
Fixes: #148741
Fixes: #148740
Fixes: #148738
Fixes: #148737
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148750)
<!-- Reviewable:end -->

DarrylWong committed:
roachtest: fix failover disk stall

This fixes the failover disk staller to be compatible with the FI
library. Specifically, it does not attempt to recover in cleanup,
as this is already handled in the FI framework.

Additionally, it temporarily disables concurrent disk stall failures
from being injected as this is an invariant of the FI framework.
Instead, they should be injected in one shot.

DarrylWong commented:
TFTR!

bors r=golgeek
